src/main/c/truffleposix/truffleposix.c

```diff
@@ -122,6 +123,15 @@ int truffleposix_select(int nread, int *readfds, int nwrite, int *writefds,
   return ret;
 }
 
+int truffleposix_utimes(const char *filename, long atime_us, long mtime_us) {
+  struct timeval timevals[2];
+  timevals[0].tv_sec = (atime_us / 1000000);
```

Comment:
`tv_sec` and `tv_usec` are both long values. By passing in microseconds and converting them, aren't you constraining the upper-bound of these fields? If so, since this is already our function, why not pass all four fields as arguments?

Reply:
Good catch!
`long` is 64-bit, so we can still express a duration of `(2**63/1e6)/3600/24/365` = 292471 years with it. NFI will complain if the value is too big, as we would pass a Bignum then.
But MRI allows to set the mtime to a value such as  `Time.at(1<<44) = year 559444`, so let's allow that too.

Reply:
Fixed.

Reply:
It's unlikely to be an issue, but it's best to avoid the pitfall. We had a similar issue at one point with the SVM and in that case it really did turn out to be a problem.